### PR TITLE
Support getting /@querystring vocabs in context

### PR DESCRIPTION
**Pull request by davisagli**
Add support for getting the `/@querystring` endpoint in a specific context, with vocabularies in that context.

This depends on https://github.com/plone/plone.app.querystring/pull/137 for getting the vocabularies in the specified context. (But the new endpoint will still work without the change in plone.app.querystring, only with the vocabularies always in the context of the portal.)

I also edited the docs for this endpoint a bit.

### CODE REVIEW

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Review by tisto — APPROVED** [review]

File: docs/source/endpoints/querystring.md [hunk 1]
```diff
@@ -1,40 +1,36 @@
...
-  Their values will be inlined in the `values` property.
-- *Operations* will be inlined as well.
-  The `operations` property will contain the list of operations as dotted names.
-  The `operators` property will contain the full definition of each of those operations supported by that field.
-- Indexes that are flagged as *sortable* are listed in a dedicated top-level property `sortable_indexes`.
+If an index uses a vocabulary, the vocabulary values are included in the `values` property.
+The vocabulary is resolved in the same context where the `/@querystring` endpoint is called (requires `plone.app.querystring >= 2.1.0`).
 
-Available options for the querystring in a Plone site can be queried by interacting with the `/@querystring` endpoint on the portal root:
+## Get Querystring Config
```
Comment (stevepiercy) [inline docs/source/endpoints/querystring.md:24]:
```suggestion
## Get `querystring` configuration
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is a solid improvement. Thank you! I have only one suggestion.

**Review by sneridagh — APPROVED** [review]

**Review by stevepiercy — APPROVED** [review]

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@davisagli: @mauritsvanrees asked me to wait with a plone.restapi 9 release. Therefore we might want to backport this feature to the plone.restapi 8.x.x branch to being able to move on...

**Comment by tisto** [comment]
@davisagli @ericof @sneridagh seems the plone.restapi build pulls in "latest" somewhere. We have a test failure now that is completely unrelated to this PR:

```
Failure in test test_get_own_user_portrait_with_svg (plone.restapi.tests.test_services_users.TestUsersEndpoint)
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/unittest/case.py", line [59](https://github.com/plone/plone.restapi/actions/runs/6262417794/job/17004534477?pr=1704#step:10:60), in testPartExecutor
    yield
  File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/unittest/case.py", line 628, in run
    testMethod()
  File "/home/runner/work/plone.restapi/plone.restapi/src/plone/restapi/tests/test_services_users.py", line 1110, in test_get_own_user_portrait_with_svg
    self.assertEqual(response.headers["Content-Type"], "image/svg+xml")
  File "/opt/hostedtoolcache/Python/3.7.17/x[64](https://github.com/plone/plone.restapi/actions/runs/6262417794/job/17004534477?pr=1704#step:10:65)/lib/python3.7/unittest/case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/unittest/case.py", line 1233, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/unittest/case.py", line [69](https://github.com/plone/plone.restapi/actions/runs/6262417794/job/17004534477?pr=1704#step:10:70)3, in fail
    raise self.failureException(msg)
AssertionError: 'text/html; charset=utf-8' != 'image/svg+xml'
- text/html; charset=utf-8
+ image/svg+xml
```

I honestly don't know what else I need to say to convince you that fetching "latest" is just plain wrong...

**Comment by tisto** [comment]
@davisagli @sneridagh @ericof I removed "latest" from the buildout some time ago. Maybe @mauritsvanrees already fixed the test and now it is the other way around (test will pass for "latest-" Plone and currently fails for the old version). Investigating (and hoping my previous comment does not make me look like an idiot now ;))...

**Comment by tisto** [comment]
@davisagli @sneridagh @ericof seems @mauritsvanrees already fixed the tests in his PR, a manual upgrade to latest = Plone 6.0.7 fixes the failing test. Everything is as it should be, sorry for the noise. :)

**Comment by mauritsvanrees** [comment]
@tisto I have changed coredev 6.0 to checkout `plone.restapi` from the main branch again.
If you want to make a release now, go ahead.

**Pull request merged**